controllers/topicPage/TopicPage.js

```diff
@@ -93,12 +96,17 @@ class TopicPage {
 
     try {
       this._validator.validateGetTopicPages({id});
-      const listBlockUser = await getListBlockUser(req.userId);
-      const listBlockDomain = await getBlockDomain(req.userId);
-      const listPostAnonymous = await getListBlockPostAnonymous(req.userId);
-      const myAnonymousUser = await UsersFunction.findAnonymousUserId(User, req.userId, {
-        raw: true
-      });
+
+      const [listBlockUser, listBlockDomain, listPostAnonymous, myAnonymousUser, isBlurredPost] =
+        await Promise.all([
+          getListBlockUser(req.userId),
+          getBlockDomain(req.userId),
+          getListBlockPostAnonymous(req.userId),
+          UsersFunction.findAnonymousUserId(User, req.userId, {
+            raw: true
+          }),
+          UserFollowUserFunction.checkIsBlurredPost(UserFollowUser, req.userId)
+        ]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change improves performance by fetching multiple data concurrently using `Promise.all()`. However, ensure that error handling is properly done for each promise. If any of these promises fail, it will cause all other promises to be rejected as well.



<!-- This is an auto-generated comment by OSS CodeRabbit -->